addons/binding/org.openhab.binding.cm11a/src/main/java/org/openhab/binding/cm11a/internal/X10Interface.java

```diff
@@ -0,0 +1,804 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.cm11a.internal;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.EOFException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TooManyListenersException;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+import org.openhab.binding.cm11a.handler.Cm11aAbstractHandler;
+import org.openhab.binding.cm11a.handler.Cm11aBridgeHandler;
+import org.openhab.binding.cm11a.handler.ReceivedDataListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gnu.io.CommPortIdentifier;
+import gnu.io.NoSuchPortException;
+import gnu.io.PortInUseException;
+import gnu.io.SerialPort;
+import gnu.io.SerialPortEvent;
+import gnu.io.SerialPortEventListener;
+import gnu.io.UnsupportedCommOperationException;
+
+/**
+ * Driver for the CM11 X10 interface.
+ *
+ *
+ * @author anthony green - Original code
```

Comment:
Ah, here we go. Might you want to write his name capitalized?
As this is code from him, it would be good to get his approval [for our DCO](https://github.com/openhab/openhab2-addons/blob/master/CONTRIBUTING.md#sign-your-work) and you must list him with an "Also-By" line in your commit comment (right above the sign off).

Reply:
Anthony Green wrote a whole openHAB-1 binding. I used this X10Interface with changes. The rest of it I started from scratch as an openHAB-2 binding. I make have copied lines from his binding. It has been over a year ago I started on this and I don't remember all of the details. As I indicated above I have been unable to contact him. I can add the Also-By line but getting his approval doesn't seem possible to me.

Reply:
No need to add any details on what you changed - it will be enough if you put something like "Bob Raker - adapted code for use in openHAB 2" or similar.